rpcs3/Emu/Cell/lv2/sys_game.cpp

```diff
@@ -3,14 +3,84 @@
 #include "Emu/Memory/vm_ptr.h"
 #include "Emu/Cell/ErrorCodes.h"
 #include "Emu/System.h"
-
+#include "Emu/system_utils.hpp"
 #include "Emu/IdManager.h"
 #include "Utilities/Thread.h"
 
 #include "sys_game.h"
 
 LOG_CHANNEL(sys_game);
 
+struct board_storage
+{
+public:
+	bool read(void* buffer)
+	{
+		if (!buffer)
+			return false;
+
+		storage.atomic_op([&](be_t<u128>& data)
+		{
+			memcpy(buffer, &data, size);
+		});
+
+		return true;
+	}
+
+	bool write(void* buffer)
+	{
+		if (!buffer)
+			return false;
+
+		storage.atomic_op([&](be_t<u128>& data)
+		{
+			memcpy(&data, buffer, size);
+		});
+
+		written = true;
+
+		return true;
+	}
+
+	board_storage()
+	{
+		storage.atomic_op([&](be_t<u128>& data)
+		{
+			if (fs::file file; file.open(file_path, fs::read))
```

Comment:
atomic_op is an opportunistic loop, it needs to be as light as possible. Also it's not needed here anyways because only one thread access it in constructor and destructor.

Reply:
I changed the atomic_op to raw for direct access.